src/SignalsQueries/LoyaltyProgram.workflow.cs

```diff
@@ -0,0 +1,38 @@
+namespace TemporalioSamples.SignalsQueries;
+
+using Microsoft.Extensions.Logging;
+using Temporalio.Workflows;
+
+[Workflow]
+public class LoyaltyProgram
+{
+    private string? userId;
+    private int points = 0;
+
+    [WorkflowRun]
+    public async Task RunAsync(string userId)
+    {
+        this.userId = userId;
+
+        // TODO dear chad, how do i await cancellation? or something short/simple that prevents from returning
```

Comment:
If you want to wait forever in and SDK, you can do a wait condition on false. In .NET this is `await Workflow.WaitConditionAsync(() => false);`